### PR TITLE
Return null instead of throwError 

### PR DESCRIPTION
**Pull request by iamtarun1993**

### CODE REVIEW

**Review by madhavmansuriya40 — APPROVED** [review]

**Review by madhavmansuriya40 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

File: src/app/core/services/permissions.service.ts [hunk 1]
```diff
@@ -2229,7 +2229,7 @@ export class PermissionsService {
           if (currentAllowedActions.allowedRouteAccess) {
             return of(currentAllowedActions);
           } else {
-            throwError('no route access');
+            return of(null);
```
Comment (Julias0) [inline src/app/core/services/permissions.service.ts:2232]: shouldnt we return the throwError?

Comment (iamtarun1993) [reply src/app/core/services/permissions.service.ts:2232]: If we return `throwError` also then the sentry also catch this and put in their console. So no point of solving this

Comment (Julias0) [reply src/app/core/services/permissions.service.ts:2232]: But why is this error coming in the first place?

File: src/app/core/services/permissions.service.ts [hunk 2]
```diff
@@ -2202,7 +2202,7 @@ export class PermissionsService {
         }
 
         return roles;
-      })
+      }),filter(roles => roles.length > 0)
```
Comment (Julias0) [inline src/app/core/services/permissions.service.ts:2205]: lets skip this approach

File: src/app/core/services/permissions.service.ts [hunk 3]
```diff
@@ -2202,7 +2202,7 @@ export class PermissionsService {
...
     );
 
     return filteredRoles$.pipe(
```
Comment (Julias0) [inline src/app/core/services/permissions.service.ts:2208]: use iif to check if fitleredRoles is null or empty array. If yes, return of(null) else lets go with this flow.

Using filter will prevent this observable from resolving which might have unintented consequences

File: src/app/core/services/permissions.service.ts [hunk 4]
```diff
@@ -2229,11 +2229,17 @@ export class PermissionsService {
           if (currentAllowedActions.allowedRouteAccess) {
             return of(currentAllowedActions);
           } else {
-            throwError('no route access');
+            return throwError('no route access');
```
Comment (abhishek1234321) [inline src/app/core/services/permissions.service.ts:2232]: is this an observable?

Comment (abhishek1234321) [inline src/app/core/services/permissions.service.ts:2232]: is this an observable?

Comment (abhishek1234321) [inline src/app/core/services/permissions.service.ts:2232]: is this an observable?

Comment (abhishek1234321) [inline src/app/core/services/permissions.service.ts:2232]: is this an observable?

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/core/services/permissions.service.ts (hunk 4 above)
Comment (iamtarun1993) [reply src/app/core/services/permissions.service.ts:2232]: yes throwErrror always return observables

**Review by Julias0 — APPROVED** [review]

**Pull request merged**